test/sagemaker_tests/pytorch/training/resources/mnist/smdataparallel_mnist.py

```diff
@@ -39,9 +39,9 @@
     import torch.distributed as dist
     import smdistributed.dataparallel.torch.torch_smddp
 
-    # set default instance type to p3.16
+    # set default instance type to p4
     if "SAGEMAKER_INSTANCE_TYPE" not in os.environ:
-        os.environ["SAGEMAKER_INSTANCE_TYPE"] = "ml.p3.16xlarge"
+        os.environ["SAGEMAKER_INSTANCE_TYPE"] = "ml.p4d.24xlarge"
```

Comment:
Why do we need P4 Instances? We might face resource capacity issues with p4.

Reply:
SMDDP library has been deprecated on p3 instances

Reply:
Thanks!